src/components/attendee/MiniProfile/components/MiniProfileModal.tsx

```diff
@@ -0,0 +1,82 @@
+import { useCallback } from "react";
+import { useHistory } from "react-router-dom";
+import { Card } from "components/attendee/Card";
+import { SpentTime } from "components/shared/SpentTime";
+import firebase from "firebase/compat/app";
+
+import { UserId, UserWithId } from "types/id";
+
+import { generateAttendeeSpaceLandingUrl } from "utils/url";
+
+import { useChatSidebarControls } from "hooks/chats/util/useChatSidebarControls";
+import { useSpaceParams } from "hooks/spaces/useSpaceParams";
+import { useIsCurrentUser } from "hooks/useIsCurrentUser";
+import { useProfileModalControls } from "hooks/useProfileModalControls";
+
+import { UserAvatar } from "components/atoms/UserAvatar";
+
+import { MiniProfileModalLinks } from "./MiniProfileModalLinks";
+
+import styles from "./MiniProfileModal.module.scss";
+
+interface MiniProfileModalProps {
+  profile: UserWithId;
+  userId: UserId;
+}
+
+export const MiniProfileModal: React.FC<MiniProfileModalProps> = ({
+  profile,
+}) => {
+  const { worldSlug, spaceSlug } = useSpaceParams();
+
+  const history = useHistory();
+
+  const { selectRecipientChat } = useChatSidebarControls();
+
+  const { closeUserProfileModal } = useProfileModalControls();
+
+  const openChat = useCallback(() => {
+    selectRecipientChat(profile);
+    closeUserProfileModal();
+  }, [selectRecipientChat, profile, closeUserProfileModal]);
+
+  const logout = useCallback(async () => {
```

Comment:
[Q] what will happen on rejection? would it be more useful to have `useAsyncFn` here and deal with the `error` field?

Reply:
changed: https://github.com/sparkletown/sparkle/pull/2953/commits/a52527d7fce74b5f15447723260cc7715c8614b3